File: Warmup/Java/CountingValleys.java
/*
 * Complete the 'countingValleys' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts following parameters:
 *  1. INTEGER steps
 *  2. STRING path
 */

public static int countingValleys(int steps, String path) {
    int valleys = 0;
    int count = 0;
    for(int i = 0; i < steps; i++) {
        if(path.charAt(i) == 'D') {
            count -= 1;
        } else {
            count += 1;
        }s
        if(count == 0 && path.charAt(i) == 'U'){
            valleys += 1;
        }            
	}
	return valleys;
}

File: Warmup/Python/counting_valleys.py
#
# Complete the 'countingValleys' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER steps
#  2. STRING path
#

def countingValleys(steps, path):
    valley = 0
    count = 0
    for i in range(0,steps):
        if path[i] == 'D':
            count -= 1
        else:
            count += 1

        if count == 0 and path[i] == 'U':
            valley += 1

    return valley
